golang/src/queries/users.go

```diff
@@ -0,0 +1,27 @@
+package queries
+
+import "database/sql"
+
+// IsUser returns whether the provided user exists in the database
+func IsUser(db *sql.DB, username string) (bool, error) {
+	var (
+		count int64
+		query = `SELECT COUNT(*) FROM ( SELECT DISTINCT id FROM users WHERE username = $1 ) AS check_user`
```

Comment:
It doesn't strictly need to go into this -- the query you're copying is doing the "take a count and check it's positive" approach too, and this is a simple rewrite -- but I think it'd be clearer to just have this query be `SELECT EXISTS(SELECT id FROM users WHERE username = $1)`, which directly returns a boolean.